### PR TITLE
[Feature][Connector-V2][Oracle-cdc]Support for oracle cdc

### LINKED ISSUE #5082: [Feature][connector-v2][oracle]Support source oracle cdc
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Support source oracle cdc 

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Carl-Zhou-CN**
## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

close #5082

### CODE REVIEW

**Comment by yinshaohua77** [comment]
Excellent

File: docs/en/connector-v2/source/Oracle-CDC.md [hunk 1]
```diff
@@ -0,0 +1,226 @@
+# Oracle CDC
```
Comment (hailin0) [inline docs/en/connector-v2/source/Oracle-CDC.md:1]: reference mysql-cdc docs

https://github.com/apache/seatunnel/blob/dev/docs/en/connector-v2/source/MySQL-CDC.md

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-oracle-e2e/src/test/java/org/apache/seatunnel/connectors/seatunnel/cdc/oracle/OracleCDCIT.java [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+import java.util.stream.Stream;
+
+import static org.awaitility.Awaitility.await;
+import static org.junit.Assert.assertNotNull;
+
+@Slf4j
+@DisabledOnContainer(
+        value = {},
+        type = {EngineType.SPARK, EngineType.FLINK},
+        disabledReason = "Currently SPARK and FLINK do not support cdc")
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-oracle-e2e/src/test/java/org/apache/seatunnel/connectors/seatunnel/cdc/oracle/OracleCDCIT.java]:
```suggestion
        type = {EngineType.SPARK},
        disabledReason = "Currently SPARK do not support cdc")
```

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-oracle-e2e/src/test/java/org/apache/seatunnel/connectors/seatunnel/cdc/oracle/OracleCDCIT.java [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+    public void startUp() throws Exception {
+        ORACLE_CONTAINER.setPortBindings(
+                Lists.newArrayList(String.format("%s:%s", ORACLE_PORT, ORACLE_PORT)));
+        log.info("Starting containers...");
+        Startables.deepStart(Stream.of(ORACLE_CONTAINER)).join();
+        log.info("Containers are started.");
+    }
+
+    @TestTemplate
+    public void test(TestContainer container) throws Exception {
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-oracle-e2e/src/test/java/org/apache/seatunnel/connectors/seatunnel/cdc/oracle/OracleCDCIT.java]: waiting for this pr merged

add MultiTableE2e

https://github.com/apache/seatunnel/pull/5620/files#diff-198d9ea0204d7713e1623ecd32a24c516808b469f8f792211e3f71b0a101b385R177

**Comment by zhilinli123** [comment]
docs/en/connector-v2/source/Oracle-CDC.md -> Also refer to the mysql-cdc.md design,Connector documentation to do some upgrades
https://github.com/apache/seatunnel/blob/dev/docs/en/connector-v2/source/MySQL-CDC.md?plain=1

**Comment by lihjChina** [comment]
To enable the CDC function for the English oracle, it is necessary to enable the additional log. I did not see any logic for enabling the configuration in the e2e.
For example, you can use the following SQL statements to enable CDC and additional logs:

ALTER DATABASE ADD SUPPLEMENTAL LOG DATA;
ALTER TABLE TEST.USER ADD SUPPLEMENTAL LOG DATA (ALL) COLUMNS;
These statements will enable CDC and additional logs for the database and the specified table.

**Comment by Carl-Zhou-CN** [comment]
> To enable the CDC function for the English oracle, it is necessary to enable the additional log. I did not see any logic for enabling the configuration in the e2e. For example, you can use the following SQL statements to enable CDC and additional logs:
> 
> ALTER DATABASE ADD SUPPLEMENTAL LOG DATA; ALTER TABLE TEST.USER ADD SUPPLEMENTAL LOG DATA (ALL) COLUMNS; These statements will enable CDC and additional logs for the database and the specified table.

The following condition will be added to the document later. E2E is enabled in the image itself.

**Comment by Carl-Zhou-CN** [comment]
@hailin0 @EricJoy2048 @lihjChina   Please help review, thank you

**Comment by yinshaohua77** [comment]
when I use this code to run oracle-cdc task, then report error :


 ERROR [o.a.s.c.s.SeaTunnel           ] [main] - Exception StackTrace:org.apache.seatunnel.core.starter.exc eption.CommandExecuteException: SeaTunnel job executed failed
        at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:192)
        at org.apache.seatunnel.core.starter.SeaTunnel.run(SeaTunnel.java:40)
        at org.apache.seatunnel.core.starter.seatunnel.SeaTunnelClient.main(SeaTunnelClient.java:34)
Caused by: org.apache.seatunnel.api.table.factory.FactoryException: ErrorCode:[API-06], ErrorDescription:[Factory initialize faile d] - Unable to create a source for identifier 'Oracle-CDC'.
        at org.apache.seatunnel.api.table.factory.FactoryUtil.createAndPrepareSource(FactoryUtil.java:100)
        at org.apache.seatunnel.engine.core.parse.MultipleTableJobConfigParser.parseSource(MultipleTableJobConfigParser.java:312)
        at org.apache.seatunnel.engine.core.parse.MultipleTableJobConfigParser.parse(MultipleTableJobConfigParser.java:180)
        at org.apache.seatunnel.engine.client.job.ClientJobExecutionEnvironment.getLogicalDag(ClientJobExecutionEnvironment.java:8 8)
        at org.apache.seatunnel.engine.client.job.ClientJobExecutionEnvironment.execute(ClientJobExecutionEnvironment.java:161)
        at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:144)
        ... 2 more
Caused by: org.apache.seatunnel.common.utils.SeaTunnelException: Can not find catalog table with factoryId [Oracle]
        at org.apache.seatunnel.api.table.catalog.CatalogTableUtil.lambda$getCatalogTables$0(CatalogTableUtil.java:129)
        at java.util.Optional.map(Optional.java:215)
        at org.apache.seatunnel.api.table.catalog.CatalogTableUtil.getCatalogTables(CatalogTableUtil.java:116)
        at org.apache.seatunnel.api.table.catalog.CatalogTableUtil.getCatalogTables(CatalogTableUtil.java:96)
        at org.apache.seatunnel.connectors.seatunnel.cdc.oracle.source.OracleIncrementalSourceFactory.lambda$createSource$0(Oracle IncrementalSourceFactory.java:103)
        at org.apache.seatunnel.api.table.factory.FactoryUtil.createAndPrepareSource(FactoryUtil.java:112)
        at org.apache.seatunnel.api.table.factory.FactoryUtil.createAndPrepareSource(FactoryUtil.java:73)
        ... 7 more

**Comment by Carl-Zhou-CN** [comment]
> when I use this code to run oracle-cdc task, then report error :
> 
> ERROR [o.a.s.c.s.SeaTunnel ] [main] - Exception StackTrace:org.apache.seatunnel.core.starter.exc eption.CommandExecuteException: SeaTunnel job executed failed at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:192) at org.apache.seatunnel.core.starter.SeaTunnel.run(SeaTunnel.java:40) at org.apache.seatunnel.core.starter.seatunnel.SeaTunnelClient.main(SeaTunnelClient.java:34) Caused by: org.apache.seatunnel.api.table.factory.FactoryException: ErrorCode:[API-06], ErrorDescription:[Factory initialize faile d] - Unable to create a source for identifier 'Oracle-CDC'. at org.apache.seatunnel.api.table.factory.FactoryUtil.createAndPrepareSource(FactoryUtil.java:100) at org.apache.seatunnel.engine.core.parse.MultipleTableJobConfigParser.parseSource(MultipleTableJobConfigParser.java:312) at org.apache.seatunnel.engine.core.parse.MultipleTableJobConfigParser.parse(MultipleTableJobConfigParser.java:180) at org.apache.seatunnel.engine.client.job.ClientJobExecutionEnvironment.getLogicalDag(ClientJobExecutionEnvironment.java:8 8) at org.apache.seatunnel.engine.client.job.ClientJobExecutionEnvironment.execute(ClientJobExecutionEnvironment.java:161) at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:144) ... 2 more Caused by: org.apache.seatunnel.common.utils.SeaTunnelException: Can not find catalog table with factoryId [Oracle] at org.apache.seatunnel.api.table.catalog.CatalogTableUtil.lambda$getCatalogTables$0(CatalogTableUtil.java:129) at java.util.Optional.map(Optional.java:215) at org.apache.seatunnel.api.table.catalog.CatalogTableUtil.getCatalogTables(CatalogTableUtil.java:116) at org.apache.seatunnel.api.table.catalog.CatalogTableUtil.getCatalogTables(CatalogTableUtil.java:96) at org.apache.seatunnel.connectors.seatunnel.cdc.oracle.source.OracleIncrementalSourceFactory.lambda$createSource$0(Oracle IncrementalSourceFactory.java:103) at org.apache.seatunnel.api.table.factory.FactoryUtil.createAndPrepareSource(FactoryUtil.java:112) at org.apache.seatunnel.api.table.factory.FactoryUtil.createAndPrepareSource(FactoryUtil.java:73) ... 7 more

This may be caused by the configuration. Check whether the permissions or connections are smooth

**Comment by EricJoy2048** [comment]
Thanks for your contribution. There are some small suggestions in the document. When enable logminer, try to complete all operations under the Oracle user as much as possible and do not switch users back and forth.

**Comment by EricJoy2048** [comment]
@hailin0  PTAL

**Comment by EricJoy2048** [comment]
I add this pr to 2.3.4 milestone.

**Comment by Carl-Zhou-CN** [comment]
> Thanks for your contribution. There are some small suggestions in the document. When enable logminer, try to complete all operations under the Oracle user as much as possible and do not switch users back and forth.

done

**Comment by yinshaohua77** [comment]
hello , Is Oracle 11g not supported ?

**Comment by Carl-Zhou-CN** [comment]
> hello , Is Oracle 11g not supported ?

@yinshaohua77 supported ,Please share the config you tested

**Comment by yinshaohua77** [comment]
thinks a lot ， here is my config, I give user  roma_logminer dba and any other right， please help me  

env {
  execution.parallelism = 1
  job.mode = "STREAMING"
}
source {
  Oracle-CDC {
    result_table_name = "stock_mysql"
    catalog = {
      factory = Oracle
    }
    username = "roma_logminer"
    password = "xxxxxx"
    database-names = ["DB220"]
    schema-names = ["ORACLE"]
    table-names = ["ORACLE.ACC_USER_BIND"]
    base-url = "jdbc:oracle:thin:roma_logminer/xxxxxx@10.10.1.98:1521:db220"
    source.reader.close.timeout = 120000
    connection.pool.size = 5
    debezium {
        database.oracle.jdbc.timezoneAsRegion = "false"
    }
  }

}

sink {
  Console {
  }
}
report error
Exception in thread "main" org.apache.seatunnel.core.starter.exception.CommandExecuteException: SeaTunnel job executed failed
        at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:192)
        at org.apache.seatunnel.core.starter.SeaTunnel.run(SeaTunnel.java:40)
        at org.apache.seatunnel.core.starter.seatunnel.SeaTunnelClient.main(SeaTunnelClient.java:34)
Caused by: org.apache.seatunnel.api.table.factory.FactoryException: ErrorCode:[API-06], ErrorDescription:[Factory initialize failed] - Unable to create a source for identifier 'Oracle-CDC'.
        at org.apache.seatunnel.api.table.factory.FactoryUtil.createAndPrepareSource(FactoryUtil.java:100)
        at org.apache.seatunnel.engine.core.parse.MultipleTableJobConfigParser.parseSource(MultipleTableJobConfigParser.java:312)
        at org.apache.seatunnel.engine.core.parse.MultipleTableJobConfigParser.parse(MultipleTableJobConfigParser.java:180)
        at org.apache.seatunnel.engine.client.job.ClientJobExecutionEnvironment.getLogicalDag(ClientJobExecutionEnvironment.java:88)
        at org.apache.seatunnel.engine.client.job.ClientJobExecutionEnvironment.execute(ClientJobExecutionEnvironment.java:161)
        at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:144)
        ... 2 more
Caused by: org.apache.seatunnel.common.utils.SeaTunnelException: Can not find catalog table with factoryId [Oracle]
        at org.apache.seatunnel.api.table.catalog.CatalogTableUtil.lambda$getCatalogTables$0(CatalogTableUtil.java:129)
        at java.util.Optional.map(Optional.java:215)
        at org.apache.seatunnel.api.table.catalog.CatalogTableUtil.getCatalogTables(CatalogTableUtil.java:116)
        at org.apache.seatunnel.api.table.catalog.CatalogTableUtil.getCatalogTables(CatalogTableUtil.java:96)
        at org.apache.seatunnel.connectors.seatunnel.cdc.oracle.source.OracleIncrementalSourceFactory.lambda$createSource$0(OracleIncrementalSourceFactory.java:103)
        at org.apache.seatunnel.api.table.factory.FactoryUtil.createAndPrepareSource(FactoryUtil.java:112)
        at org.apache.seatunnel.api.table.factory.FactoryUtil.createAndPrepareSource(FactoryUtil.java:73)

**Comment by Carl-Zhou-CN** [comment]
> thinks a lot ， here is my config, I give user roma_logminer dba and any other right， please help me
> 
> env { execution.parallelism = 1 job.mode = "STREAMING" } source { Oracle-CDC { result_table_name = "stock_mysql" catalog = { factory = Oracle } username = "roma_logminer" password = "xxxxxx" database-names = ["DB220"] schema-names = ["ORACLE"] table-names = ["ORACLE.ACC_USER_BIND"] base-url = "jdbc:oracle:thin:roma_logminer/xxxxxx@10.10.1.98:1521:db220" source.reader.close.timeout = 120000 connection.pool.size = 5 debezium { database.oracle.jdbc.timezoneAsRegion = "false" } }
> 
> }
> 
> sink { Console { } } report error Exception in thread "main" org.apache.seatunnel.core.starter.exception.CommandExecuteException: SeaTunnel job executed failed at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:192) at org.apache.seatunnel.core.starter.SeaTunnel.run(SeaTunnel.java:40) at org.apache.seatunnel.core.starter.seatunnel.SeaTunnelClient.main(SeaTunnelClient.java:34) Caused by: org.apache.seatunnel.api.table.factory.FactoryException: ErrorCode:[API-06], ErrorDescription:[Factory initialize failed] - Unable to create a source for identifier 'Oracle-CDC'. at org.apache.seatunnel.api.table.factory.FactoryUtil.createAndPrepareSource(FactoryUtil.java:100) at org.apache.seatunnel.engine.core.parse.MultipleTableJobConfigParser.parseSource(MultipleTableJobConfigParser.java:312) at org.apache.seatunnel.engine.core.parse.MultipleTableJobConfigParser.parse(MultipleTableJobConfigParser.java:180) at org.apache.seatunnel.engine.client.job.ClientJobExecutionEnvironment.getLogicalDag(ClientJobExecutionEnvironment.java:88) at org.apache.seatunnel.engine.client.job.ClientJobExecutionEnvironment.execute(ClientJobExecutionEnvironment.java:161) at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:144) ... 2 more Caused by: org.apache.seatunnel.common.utils.SeaTunnelException: Can not find catalog table with factoryId [Oracle] at org.apache.seatunnel.api.table.catalog.CatalogTableUtil.lambda$getCatalogTables$0(CatalogTableUtil.java:129) at java.util.Optional.map(Optional.java:215) at org.apache.seatunnel.api.table.catalog.CatalogTableUtil.getCatalogTables(CatalogTableUtil.java:116) at org.apache.seatunnel.api.table.catalog.CatalogTableUtil.getCatalogTables(CatalogTableUtil.java:96) at org.apache.seatunnel.connectors.seatunnel.cdc.oracle.source.OracleIncrementalSourceFactory.lambda$createSource$0(OracleIncrementalSourceFactory.java:103) at org.apache.seatunnel.api.table.factory.FactoryUtil.createAndPrepareSource(FactoryUtil.java:112) at org.apache.seatunnel.api.table.factory.FactoryUtil.createAndPrepareSource(FactoryUtil.java:73)

`env {
  execution.parallelism = 1
  job.mode = "STREAMING"
  checkpoint.interval = 5000
}

source {
  Oracle-CDC {
    result_table_name = "customers"
    username = "system"
    password = "oracle"
    database-names = ["XE"]
    schema-names = ["DEBEZIUM"]
    table-names = ["XE.DEBEZIUM.FULL_TYPES"]
    base-url = "jdbc:oracle:thin:system/oracle@oracle-host:1521:xe"
    source.reader.close.timeout = 120000
    connection.pool.size = 5
    debezium {
       # log.mining.strategy = "online_catalog"
       # log.mining.continuous.mine = true
        database.oracle.jdbc.timezoneAsRegion = "false"
    }
  }
}

transform {
}

sink {
  console {
    source_table_name = "customers"
  }
Jdbc {
  source_table_name = "customers"
  driver = "oracle.jdbc.driver.OracleDriver"
  url = "jdbc:oracle:thin:system/oracle@oracle-host:1521:xe"
  user = "system"
  password = "oracle"
  generate_sink_sql = true
  database = "XE"
  table = "DEBEZIUM.SINK_FULL_TYPES"
  batch_size = 1
  primary_keys = ["ID"]
}
}`
@yinshaohua77  Please refer to this configuration to update your config

**Comment by Carl-Zhou-CN** [comment]
> > thinks a lot ， here is my config, I give user roma_logminer dba and any other right， please help me
> > env { execution.parallelism = 1 job.mode = "STREAMING" } source { Oracle-CDC { result_table_name = "stock_mysql" catalog = { factory = Oracle } username = "roma_logminer" password = "xxxxxx" database-names = ["DB220"] schema-names = ["ORACLE"] table-names = ["ORACLE.ACC_USER_BIND"] base-url = "jdbc:oracle:thin:roma_logminer/xxxxxx@10.10.1.98:1521:db220" source.reader.close.timeout = 120000 connection.pool.size = 5 debezium { database.oracle.jdbc.timezoneAsRegion = "false" } }
> > }
> > sink { Console { } } report error Exception in thread "main" org.apache.seatunnel.core.starter.exception.CommandExecuteException: SeaTunnel job executed failed at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:192) at org.apache.seatunnel.core.starter.SeaTunnel.run(SeaTunnel.java:40) at org.apache.seatunnel.core.starter.seatunnel.SeaTunnelClient.main(SeaTunnelClient.java:34) Caused by: org.apache.seatunnel.api.table.factory.FactoryException: ErrorCode:[API-06], ErrorDescription:[Factory initialize failed] - Unable to create a source for identifier 'Oracle-CDC'. at org.apache.seatunnel.api.table.factory.FactoryUtil.createAndPrepareSource(FactoryUtil.java:100) at org.apache.seatunnel.engine.core.parse.MultipleTableJobConfigParser.parseSource(MultipleTableJobConfigParser.java:312) at org.apache.seatunnel.engine.core.parse.MultipleTableJobConfigParser.parse(MultipleTableJobConfigParser.java:180) at org.apache.seatunnel.engine.client.job.ClientJobExecutionEnvironment.getLogicalDag(ClientJobExecutionEnvironment.java:88) at org.apache.seatunnel.engine.client.job.ClientJobExecutionEnvironment.execute(ClientJobExecutionEnvironment.java:161) at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:144) ... 2 more Caused by: org.apache.seatunnel.common.utils.SeaTunnelException: Can not find catalog table with factoryId [Oracle] at org.apache.seatunnel.api.table.catalog.CatalogTableUtil.lambda$getCatalogTables$0(CatalogTableUtil.java:129) at java.util.Optional.map(Optional.java:215) at org.apache.seatunnel.api.table.catalog.CatalogTableUtil.getCatalogTables(CatalogTableUtil.java:116) at org.apache.seatunnel.api.table.catalog.CatalogTableUtil.getCatalogTables(CatalogTableUtil.java:96) at org.apache.seatunnel.connectors.seatunnel.cdc.oracle.source.OracleIncrementalSourceFactory.lambda$createSource$0(OracleIncrementalSourceFactory.java:103) at org.apache.seatunnel.api.table.factory.FactoryUtil.createAndPrepareSource(FactoryUtil.java:112) at org.apache.seatunnel.api.table.factory.FactoryUtil.createAndPrepareSource(FactoryUtil.java:73)

`> 
> env {
> 
> # You can set engine configuration here
> execution.parallelism = 1 job.mode = "STREAMING" checkpoint.interval = 5000 }
> 
> source {
> 
> # This is a example source plugin **only for test and demonstrate the feature source plugin**
> Oracle-CDC { result_table_name = "customers" username = "system" password = "oracle" database-names = ["XE"] schema-names = ["DEBEZIUM"] table-names = ["XE.DEBEZIUM.FULL_TYPES"] base-url = "jdbc:oracle:thin:system/oracle@oracle-host:1521:xe" source.reader.close.timeout = 120000 connection.pool.size = 5 debezium { # log.mining.strategy = "online_catalog" # log.mining.continuous.mine = true database.oracle.jdbc.timezoneAsRegion = "false" } } }
> 
> transform { }
> 
> sink { console { source_table_name = "customers" } Jdbc { source_table_name = "customers" driver = "oracle.jdbc.driver.OracleDriver" url = "jdbc:oracle:thin:system/oracle@oracle-host:1521:xe" user = "system" password = "oracle" generate_sink_sql = true database = "XE" table = "DEBEZIUM.SINK_FULL_TYPES" batch_size = 1 primary_keys = ["ID"] } }`
 @yinshaohua77 Please refer to this configuration to update your config

**Comment by EricJoy2048** [comment]
@hailin0  PTAL

**Review by hailin0 — APPROVED** [review]
LGTM

**Review by EricJoy2048 — APPROVED** [review]

**Comment by mtzcsc** [comment]
xception in thread "main" org.apache.seatunnel.core.starter.exception.CommandExecuteException: SeaTunnel job executed failed
        at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:199)
        at org.apache.seatunnel.core.starter.SeaTunnel.run(SeaTunnel.java:40)
        at org.apache.seatunnel.core.starter.seatunnel.SeaTunnelClient.main(SeaTunnelClient.java:34)
Caused by: org.apache.seatunnel.api.table.factory.FactoryException: ErrorCode:[API-06], ErrorDescription:[Factory initialize failed] - Unable to create a source for identifier 'Oracle-CDC'.
        at org.apache.seatunnel.api.table.factory.FactoryUtil.createAndPrepareSource(FactoryUtil.java:100)
        at org.apache.seatunnel.engine.core.parse.MultipleTableJobConfigParser.parseSource(MultipleTableJobConfigParser.java:320)
        at org.apache.seatunnel.engine.core.parse.MultipleTableJobConfigParser.parse(MultipleTableJobConfigParser.java:181)
        at org.apache.seatunnel.engine.client.job.ClientJobExecutionEnvironment.getLogicalDag(ClientJobExecutionEnvironment.java:88)
        at org.apache.seatunnel.engine.client.job.ClientJobExecutionEnvironment.execute(ClientJobExecutionEnvironment.java:161)
        at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:146)
        ... 2 more
Caused by: org.apache.seatunnel.common.utils.SeaTunnelException: Can not find catalog table with factoryId [Oracle]
        at org.apache.seatunnel.api.table.catalog.CatalogTableUtil.lambda$getCatalogTables$0(CatalogTableUtil.java:129)
        at java.base/java.util.Optional.map(Optional.java:260)
        at org.apache.seatunnel.api.table.catalog.CatalogTableUtil.getCatalogTables(CatalogTableUtil.java:116)
        at org.apache.seatunnel.api.table.catalog.CatalogTableUtil.getCatalogTables(CatalogTableUtil.java:96)
        at org.apache.seatunnel.connectors.seatunnel.cdc.oracle.source.OracleIncrementalSourceFactory.lambda$createSource$1(OracleIncrementalSourceFactory.java:108)
        at org.apache.seatunnel.api.table.factory.FactoryUtil.createAndPrepareSource(FactoryUtil.java:112)
        at org.apache.seatunnel.api.table.factory.FactoryUtil.createAndPrepareSource(FactoryUtil.java:73)
        ... 7 more

      env {
        execution.parallelism = 1
        job.mode = "STREAMING"
        job.name = 'cdc'
        # 10秒检查一次，可以适当加大这个值
        checkpoint.interval = 10000
      }

      # 配置数据源
      source {
        Oracle-CDC {
         #result_table_name = "customers"
         username = "sys as sysdba"
         password = "123456"
         database-names = ["ORCL"]
         #schema-names = ["c##test"]
         table-names = ["sys.kwq"]
         base-url = "jdbc:oracle:thin:@172.20.10.12:1521:orcl"
         source.reader.close.timeout = 120000
  }
      }

      # 配置目标库
      sink {
        starrocks {
          batch_max_rows=10240
          nodeUrls = ["172.20.10.13:8030"]    ## 插入的  node  节点
          base-url = "jdbc:mysql://172.20.10.13:9030/"   ##   插入库的jdbbc链接
          username = "root"
          password = "123456"
          database = "cdc"
          data_save_mode="APPEND_DATA"
          enable_upsert_delete = true
          #table = "test"
      }
    }
~
~
请教

**Pull request merged**